src/main/java/com/monitorjbl/xlsx/StreamingReader.java

```diff
@@ -111,11 +127,57 @@ private void handleEvent(XMLEvent event) throws SAXException {
         currentCell = new StreamingCell(CellReference.convertColStringToIndex(coord[0]), Integer.parseInt(coord[1]) - 1);
 
         Attribute type = startElement.getAttributeByName(new QName("t"));
+/*ForgetMeNot--ibell--20150911--old code which didn't determine numeric vs date data types.
         if(type != null) {
           currentCell.setType(type.getValue());
         }
+*/
+        //
+        // This block will determine the value type and set the appropriate
+        // XSSF data type (used to format numeric cell values).
+        //
+        final String typeValue = (type != null) ? type.getValue() : "";
```

Comment:
I understand, and I don't want to dismiss your suggestion. I just don't want it merged in as-is, it needs to be refactored a bit to fit into the code properly. I can decline the PR and implement it myself if you want, but it's a common courtesy on GitHub to preserve the committer's contribution in the history of the repo. It's up to you, really; if you want to take the time to improve the submission I'm happy to accept it.